addons/account/wizard/account_move_send.py

```diff
@@ -0,0 +1,464 @@
+
+    @api.model
+    def _download(self, move):
+        """ Download the PDF. """
+        return {
+            'type': 'ir.actions.act_url',
+            'url': f'/web/content/{move.invoice_pdf_report_id.id}?download=true',
+            'close_on_report_download': True,
+        }
+
+    @api.model
+    def _send_mail(self, mail_template, move, **kwargs):
+        """ Send the journal entries passed as parameter by mail. """
+        partner_ids = kwargs.get('partner_ids', [])
+
+        move\
+            .with_context(
+                no_new_invoice=True,
+                mail_notify_author=self.env.user.partner_id.id in partner_ids,
+                mailing_document_based=True)\
+            .message_post(
+                message_type='comment',
+                **kwargs,
+                **{
+                    'email_layout_xmlid': 'mail.mail_notification_layout_with_responsible_signature',
+                    'email_add_signature': not mail_template,
+                    'mail_auto_delete': mail_template.auto_delete,
+                    'mail_server_id': mail_template.mail_server_id.id,
+                    'reply_to_force_new': False,
+                },
+            )
+        move.is_move_sent = True
+
+    def _send_mails(self, moves):
+        subtype = self.env.ref('mail.mt_comment')
+        mail_template = self.mail_template_id
+
+        for move in moves:
+            attachment_ids = self._get_mail_attachments_ids(mail_template, move)
+            mail_lang = self.mail_lang or self._get_default_lang(mail_template, move)
+            partners = self.mail_partner_ids or self._get_default_mail_partners(mail_template, mail_lang, move)
+            body = self.mail_body or self._get_default_mail_body(mail_template, mail_lang, move)
+            subject = self.mail_subject or self._get_default_mail_subject(mail_template, mail_lang, move)
+            email_from = self._get_default_email_from(mail_template, mail_lang, move)
+
+            self._send_mail(
+                mail_template,
+                move,
+                body=body,
+                subject=subject,
+                email_from=email_from,
+                subtype_id=subtype.id,
+                partner_ids=partners.ids,
+                attachment_ids=attachment_ids,
+            )
+
+    def _generate_documents_success_hook(self, moves, from_cron):
+        send_mail = self.enable_send_mail and self.checkbox_send_mail
+
+        # Send mail.
+        if send_mail:
+            self._send_mails(moves)
+
+        return moves
+
+    def _generate_documents_failed_hook(self, errors, from_cron):
+        for move, error in errors:
+            if from_cron:
+                move\
+                    .with_context(no_new_invoice=True)\
+                    .message_post(body=error)
+            else:
+                raise UserError(error)
+
+    def _link_document(self, invoice, prepared_data):
+        """ Link prepared_data to the record
+        """
```

Comment:
One liner docstrings should be on... one line
```suggestion
        """Link prepared_data to the record."""
```